### PR TITLE
Add custom tolerations documentation

### PR DESCRIPTION
**Pull request by jvc5546**
App PR: https://github.com/okteto/app/pull/8029

### CODE REVIEW

File: src/content/self-hosted/helm-configuration.mdx [hunk 1]
```diff
@@ -155,6 +155,7 @@ The build service. It's used in combination with `okteto build` to build contain
 - `persistence.storageClass`: The storage class of the persistence volume attached to every buildkit pod.
 - `persistence.size`: The size of the persistence volume attached to every buildkit pod. Defaults to `750Gi`.
 - `persistence.cache`: The size (in Mi) of the buildkit cache to store image caches. It should be around 30Gi smaller than `storage.size`. Defaults to 500Gi.
+- `tolerations`: List of tolerations to be added to the buildkit and registry services. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.buildPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.buildPool`. If not set, the buildkit Pods will inherit the tolerations list set in [`globals.tolerations.okteto`](self-hosted/helm-configuration.mdx#globals).
```
Comment (pchico83) [inline src/content/self-hosted/helm-configuration.mdx:158]: This should only apply yo `buildkit`, not to the registry

File: src/content/self-hosted/helm-configuration.mdx [hunk 2]
```diff
@@ -418,6 +422,25 @@ globals:
   priorityClassName: okteto-priority
 ```
 
+- `tolerations`: Tolerations to be added to Pods that are either under the `okteto` or `dev` categories, as shown below.
+  - `okteto`: A list of tolerations to be added to Pods related to Okteto components (i.e. Okteto api), which run in the `okteto` namespaces. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.oktetoPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.oktetoPool`.
```
Comment (pchico83) [inline src/content/self-hosted/helm-configuration.mdx:426]:
```suggestion
  - `okteto`: A list of tolerations to be added to Pods related to Okteto components (i.e. Okteto api), which run in the `okteto` namespace. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.oktetoPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.oktetoPool`.
```

File: src/content/self-hosted/helm-configuration.mdx [hunk 3]
```diff
@@ -418,6 +422,25 @@ globals:
...
+  - `dev`: A list of tolerations to be added to Pods related to development workloads (i.e. user application Pods), which run in personal namespaces. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.devPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.devPool`.
```
Comment (pchico83) [inline src/content/self-hosted/helm-configuration.mdx:427]:
```suggestion
  - `dev`: A list of tolerations to be added to Pods created by user applications, which run in namespaces managed by Okteto. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.devPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.devPool`.
```

File: src/content/self-hosted/helm-configuration.mdx [hunk 4]
```diff
@@ -812,12 +835,13 @@ secret:
 
 ### tolerations
 
-Indicates [tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) for the okteto components.
-Define the label and taint `okteto-node-pool` on your worker nodes to match these values.
+Indicates [tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) and [nodeSelector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) for the okteto components.
```
Comment (pchico83) [inline src/content/self-hosted/helm-configuration.mdx:838]:
```suggestion
Indicates [tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) and [nodeSelector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) for different kind of Pods in the cluster where Okteto is installed.
```

**Review by pchico83 — CHANGES_REQUESTED** [review]

File: src/content/self-hosted/helm-configuration.mdx (hunk 1 above)
Comment (jvc5546) [reply src/content/self-hosted/helm-configuration.mdx:158]: In the chart, the registry-deployment does inherit tolerations from the [`buildkit.tolerations`](https://github.com/okteto/app/blob/05abcb11331d3c3118c3d16bd7769a85deeea019/chart/okteto/templates/registry-deployment.yaml#L77) and was also in the [Custom Tolerations spec](https://www.notion.so/okteto/Custom-Tolerations-in-Chart-61b648a42e7a4d5b89904d6d9823fb2a?pvs=4#e5fc5b24f2a64f5cbbc976601df98232), but let me know if we are no longer treating them as together under the same Build pool, and the `registry` should only inherit from the Okteto tolerations. 

It should be a relatively quick change in the chart and chart tests.

Comment (jvc5546) [reply src/content/self-hosted/helm-configuration.mdx:158]: PR for code changes to match documentation changes: https://github.com/okteto/app/pull/8073

Comment (jvc5546) [reply src/content/self-hosted/helm-configuration.mdx]: Fixed the documentation to reflect that the registry will no longer inherit tolerations from the `buildkit.tolerations` helm chart value.

**Review by pchico83 — APPROVED** [review]

**Pull request merged**